EndlessClient/HUD/Inventory/InventoryPanelItem.cs

```diff
@@ -244,10 +247,36 @@ protected override void OnDrawControl(GameTime gameTime)
                 }
             }
 
-            _spriteBatch.Draw(_itemGraphic, DrawPositionWithParentOffset, Color.FromNonPremultiplied(255, 255, 255, _beingDragged ? 128 : 255));
+            if (_beingDragged)
+            {
+                // slot based on current mouse position if being dragged
+                var currentSlot = GetCurrentSlotBasedOnPosition();
+                var drawPosition = GetPosition(currentSlot) + (_beingDragged ? _oldOffset : ImmediateParent.DrawPositionWithParentOffset);
 
-            _spriteBatch.End();
+                if (!dragItemPositioned)
+                {
+                    if (InventoryGridArea.Contains(DrawArea.WithPosition(drawPosition)))
+                    {
+                        dragItemPositioned = true;
+                    }
+                }
 
+                if (dragItemPositioned)
+                {
+                    _spriteBatch.Draw(_itemGraphic, DrawPositionWithParentOffset, Color.FromNonPremultiplied(255, 255, 255, _beingDragged ? 128 : 255));
+                }
+
+                if (InventoryGridArea.Contains(DrawArea.WithPosition(drawPosition)))
+                {
+                    _spriteBatch.Draw(_itemGraphic, _beingDragged ? (DrawPositionWithParentOffset) : DrawPositionWithParentOffset, Color.FromNonPremultiplied(255, 255, 255, _beingDragged ? 128 : 255));
```

Comment:
`_beingDragged` ternary expression isn't doing anything. And, `_beingDragged` is always `true` here